File: Array/yanghao/java/Algorithm6_24.java
package jdbc.mvc.interfaces;

import java.util.LinkedHashSet;
import java.util.Set;

public class Algorithm6_24 {

	public static boolean hashSolution(ListNode head) {
		Set<ListNode> set = new LinkedHashSet<>();
		while (head != null) {
			if (set.contains(head)) {
				return true;
			}
			set.add(head);
			head = head.next;
		}
		return false;
	}
}

class ListNode {
	int val;
	ListNode next;

	ListNode(int x) {
		val = x;
		next = null;
	}
}

File: Array/yanghao/kotlin/Algorithm6_24.kt
package staudy

class Algorithm6_24 {

    fun hashSolution(head: ListNode?):Boolean{
        var a = head
        val set = LinkedHashSet<ListNode>()
        while (a != null){
           if (set.contains(a)){
               return true
           }
            set.add(a)
            a = a.next
        }

        return false
    }
}

class ListNode(var `val`: Int, var next: ListNode?) {

}